modules/outbound-command-event-handler/src/domain/bulk_batch_entity.ts

```diff
@@ -0,0 +1,208 @@
+            fspId: party.partyIdInfo.fspId,
+            extensionList: party.partyIdInfo.extensionList?.extension
+        }
+    }
+
+    static CreateEmptyBatch(
+        bulkTransactionEntity: BulkTransactionEntity
+    ): BulkBatchEntity {
+
+        const bulkQuoteId = randomUUID();
+        const bulkTransferId = randomUUID();
+
+        const initialState: BulkBatchState = {
+            id: randomUUID(),
+            state: BulkBatchInternalState.CREATED,
+            bulkId: bulkTransactionEntity.id,
+            bulkQuoteId,
+            bulkTransferId,
+            bulkQuotesRequest: {
+                homeTransactionId: bulkTransactionEntity.bulkHomeTransactionID,
+                bulkQuoteId,
+                from: BulkBatchEntity._convertPartyToFrom(bulkTransactionEntity.from),
+                individualQuotes: [],
+                extensions: bulkTransactionEntity.extensions
+            },
+            bulkTransfersRequest: {
+                homeTransactionId: bulkTransactionEntity.bulkHomeTransactionID,
+                bulkTransferId,
+                from: BulkBatchEntity._convertPartyToFrom(bulkTransactionEntity.from),
+                individualTransfers: [],
+                extensions: bulkTransactionEntity.extensions
+            },
+            quoteIdReferenceIdMap: {},
+            transferIdReferenceIdMap: {},
+            created_at: Date.now(),
+            updated_at: Date.now(),
+            version: 1,
+        };
+        return new BulkBatchEntity(initialState);
+    }
+
+    addIndividualQuote(individualQuote: SDKSchemeAdapter.Outbound.V2_0_0.Types.individualQuote, referenceId: string) {
+        this._state.bulkQuotesRequest.individualQuotes.push(individualQuote);
+        this._state.quoteIdReferenceIdMap[individualQuote.quoteId] = referenceId;
+    }
+
+    addIndividualTransfer(individualTransfer: SDKSchemeAdapter.Outbound.V2_0_0.Types.individualTransfer, referenceId: string) {
+        this._state.bulkTransfersRequest.individualTransfers.push(individualTransfer);
+        this._state.transferIdReferenceIdMap[individualTransfer.transferId] = referenceId;   
+    }
+
+    getReferenceIdForQuoteId(quoteId: string) : string {
+        return this._state.quoteIdReferenceIdMap[quoteId];
+    }
+
+    getReferenceIdForTransferId(transferId: string) : string {
+        return this._state.transferIdReferenceIdMap[transferId];
+    }
+
+    get state() {
+        return this._state.state;
+    }
+
+    setState(state: BulkBatchInternalState) {
+        this._state.state = state;
+    }
+
+    setBulkQuotesResponse(response: SDKSchemeAdapter.Outbound.V2_0_0.Types.bulkQuoteResponse) {
+        this._state.bulkQuotesResponse = response;
+    }
+
+    /* eslint-disable-next-line @typescript-eslint/no-useless-constructor */
+    constructor(initialState: BulkBatchState) {
+        // Commenting the validation in the constuctor to allow creation of bulkQuotes without any individualQuotes items
```

Comment:
This to me is abusing our types, surely if we can create bulkQuotes without individualQuotes, then the types should reflect this?

Reply:
@mdebarros , we are not doing validation on the batch creation but we are doing validation of bulkQuotes after preparing the bulkQuotes request.

Reply:
Done